Dockerfile

```diff
@@ -292,6 +292,19 @@ WORKDIR /home/builder/rust
 RUN \
   dir=build/cache/$(awk '/^date:/ { print $2 }' src/stage0.txt); \
   mkdir -p $dir && mv ../*.xz $dir
+
+RUN \
+  for libc in gnu musl ; do \
+    cp src/librustc_target/spec/${ARCH}_{unknown,${VENDOR}}_linux_${libc}.rs ; \
+    sed -i -e 's@target_vendor: "unknown"@target_vendor: "'${VENDOR}'"@' \
+      src/librustc_target/spec/${ARCH}_${VENDOR}_linux_${libc}.rs ; \
+    sed -i -e '/("'${ARCH}-unknown-linux-${libc}'", .*),/a("'${ARCH}-${VENDOR}-linux-${libc}'", '${ARCH}_${VENDOR}_linux_${libc}'),' \
+      src/librustc_target/spec/mod.rs ; \
```

Comment:
Maybe worthy of an explanation for the less gifted?

Reply:
I've added an explanation; let me know if it suffices or if I should expand on the details.

What this amounts to is the cheapest way I could find to copy existing Rust targets, change the "vendor" field, and integrate them into `rustc` so we could use them.

Reply:
That's great! 🎓